app/components/InteractContractModal/DeployContract.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+import React, { Component } from 'react';
+import TextField from '../TextField';
+import TezosNumericInput from '../TezosNumericInput/';
+import { ms } from '../../styles/helpers';
+import Fees from '../Fees';
+import PasswordInput from '../PasswordInput';
+import TezosAmount from '../TezosAmount';
+import TezosAddress from '../TezosAddress';
+import DeployLedgerConfirmationModal from '../DeployLedgerConfirmationModal';
+
+import { OPERATIONFEE } from '../../constants/LowFeeValue';
+
+import { openLinkToBlockExplorer } from '../../utils/general';
+
+import {
+  MainContainer,
+  TabContainer,
+  InputAddressContainer,
+  ParametersContainer,
+  DeployAddressContainer,
+  DeployAddressLabel,
+  DeployAddressContent,
+  SpaceBar,
+  RowContainer,
+  ColContainer,
+  AmountContainer,
+  FeeContainer,
+  PasswordButtonContainer,
+  InvokeButton,
+  UseMax
+} from './style';
+
+const utez = 1000000;
+
+type Props = {
+  isLoading: boolean,
+  isLedger: boolean,
+  averageFees: object,
+  addresses: List,
+  originateContract: () => {},
+  onClose: () => {},
+  t: () => {}
+};
+
+const defaultState = {
+  gas: 0,
+  storage: 0,
+  fee: 1420,
+  amount: '',
+  parameters: '',
+  passPhrase: '',
+  isShowedPwd: false,
+  isOpenLedgerConfirm: false,
+  michelsonCode: ''
+};
+
+class DeployContract extends Component<Props> {
+  props: Props;
+  state = {
+    ...defaultState,
+    fee: this.props.averageFees.low
+  };
+
+  onUseMax = () => {
+    const { fee, gas, storage } = this.state;
+    const { addresses } = this.props;
+    const { balance } = addresses[0];
+    const max = balance - fee - gas - storage;
+    let amount = '0';
+    if (max > 0) {
+      amount = (max / utez).toFixed(6);
+    }
+    this.setState({ amount });
+  };
+
+  updatePassPhrase = passPhrase => this.setState({ passPhrase });
+
+  onLedgerConfirmation = val => {
+    this.setState({ isOpenLedgerConfirm: val });
+  };
+
+  openLink = element => openLinkToBlockExplorer(element);
+
+  onEnterPress = (keyVal, isDisabled) => {
+    if (keyVal === 'Enter' && isDisabled) {
+      this.onDeployOperation();
+    }
+  };
+
+  onDeployOperation = async () => {
+    const {
+      originateContract,
+      setIsLoading,
+      isLedger,
+      addresses,
+      onClose
+    } = this.props;
+
+    const {
+      passPhrase,
+      amount,
+      gas,
+      storage,
+      fee,
+      michelsonCode,
+      parameters
+    } = this.state;
+
+    setIsLoading(true);
+
+    if (isLedger) {
+      this.onLedgerConfirmation(true);
+    }
+
+    const { pkh } = addresses[0];
+    const initStorage = parameters ? JSON.parse(parameters) : {};
+    const initCode = michelsonCode ? JSON.parse(michelsonCode) : [];
+    const bakerAddress = 'tz3WXYtyDUNL91qfiCJtVUX746QpNv5i5ve5';
+
+    const isOperationCompleted = await originateContract(
+      bakerAddress,
```

Comment:
We no longer need bakerAddress.

Please remove
`const bakerAddress = 'tz3WXYtyDUNL91qfiCJtVUX746QpNv5i5ve5';`
and send `undefined` as the first parameter of originateContract